neevaco/neeva-android
# Ensure FirstRun initialization works properly

Some SharedPreference values should only be set when the user starts the app for the very first time (when `FirstRunDone = false`). @dan-alcantara-neeva pointed out that the onboarding ad-blocking popup was still being shown even though he had seen it already. 

This PR ensures that the first-time initializations happen in a cohesive and clear place.  

### How was this tested?

### Manual test cases
* Did the FirstRun for the first time.
* Navigated to buzzfeed and saw the onboarding popup.
* Logged in (and therefore launched the `WelcomeFlowActivity`)
* Navigated to buzzfeed and did NOT see the onboarding popup.
